app/models/visit.rb

```diff
@@ -1,6 +1,10 @@
 class Visit < ActiveRecord::Base
   extend FreshnessCalculations
   include PrincipalVisitor
+
+  attr_writer :book_to_nomis_opt_out
```

Comment:
We need to get the opt out logic to the booking responder acceptor class, the part that decides to make the api call or not, once is in that part of the code it only has access to the visit. Have you got any suggestions on how to improve this? I agree that is not ideal.